cmd/kops/rollingupdatecluster.go

```diff
@@ -399,20 +404,21 @@ func RunRollingUpdateCluster(f *util.Factory, out io.Writer, options *RollingUpd
 		}
 	}
 	d := &instancegroups.RollingUpdateCluster{
-		MasterInterval:    options.MasterInterval,
-		NodeInterval:      options.NodeInterval,
-		BastionInterval:   options.BastionInterval,
-		Interactive:       options.Interactive,
-		Force:             options.Force,
-		Cloud:             cloud,
-		K8sClient:         k8sClient,
-		ClusterValidator:  clusterValidator,
-		FailOnDrainError:  options.FailOnDrainError,
-		FailOnValidate:    options.FailOnValidate,
-		CloudOnly:         options.CloudOnly,
-		ClusterName:       options.ClusterName,
-		PostDrainDelay:    options.PostDrainDelay,
-		ValidationTimeout: options.ValidationTimeout,
+		MasterInterval:       options.MasterInterval,
+		NodeInterval:         options.NodeInterval,
+		BastionInterval:      options.BastionInterval,
+		Interactive:          options.Interactive,
+		Force:                options.Force,
+		Cloud:                cloud,
+		K8sClient:            k8sClient,
+		ClusterValidator:     clusterValidator,
+		FailOnDrainError:     options.FailOnDrainError,
+		FailOnValidate:       options.FailOnValidate,
+		CloudOnly:            options.CloudOnly,
+		ClusterName:          options.ClusterName,
+		PostDrainDelay:       options.PostDrainDelay,
+		ValidationTimeout:    options.ValidationTimeout,
+		PreValidationTimeout: options.PreValidationTimeout,
 		// TODO should we expose this to the UI?
 		ValidateTickDuration: 30 * time.Second,
```

Comment:
tick could be faster for pre-validation-timeout. However, I decided to not make new variable for that. So lets start with 30 seconds tick

Reply:
I have a vision of validation setting watches instead of polling, at least for the API server checks.